packages/cdk-assets/lib/publishing.ts

```diff
@@ -0,0 +1,114 @@
+import { AssetManifest, IManifestEntry } from './asset-manifest';
+import { IAws } from "./aws-operations";
+import { makeAssetHandler } from "./handlers";
+
+export interface IPublishProgressListener {
+  onAssetStart(event: IPublishProgress): void;
+  onAssetEnd(event: IPublishProgress): void;
+  onEvent(event: IPublishProgress): void;
+  onError(event: IPublishProgress): void;
+}
+
+export interface IPublishProgress {
+  readonly message: string;
+  readonly currentAsset?: IManifestEntry;
+  readonly percentComplete: number;
+
+  /**
+   * Abort the current publishing operation
```

Comment:
Indicates that this will only request an abort which will only happen as soon as the current operation finishes.